src/main/java/org/springframework/data/elasticsearch/repository/support/SimpleReactiveElasticsearchRepository.java

```diff
@@ -169,7 +196,14 @@ public SimpleReactiveElasticsearchRepository(ElasticsearchEntityInformation<T, I
 	public Mono<Void> deleteAll(Publisher<? extends T> entityStream) {
 
 		Assert.notNull(entityStream, "EntityStream must not be null!");
-		return Flux.from(entityStream).flatMap(this::delete).then();
+		return deleteById(Flux.from(entityStream) //
+				.handle((o, synchronousSink) -> {
+
+					ID id = entityInformation.getId(o);
```

Comment:
An entity without an Id value is an unexpected situation and should be handled as error.